code/game/objects/items/weapons/grenades/holy_grenade.dm

```diff
@@ -0,0 +1,25 @@
+/obj/item/weapon/grenade/holy
+	name = "holy hand grenade"
+	desc = "One of the many sacred relics made for blowing thine enemies to tiny bits."
+	icon_state = "holy_grenade"
+	det_time = 3 SECONDS
+	armsound = 'sound/weapons/vampkiller.ogg'
+
+/obj/item/weapon/grenade/holy/attackby(obj/item/weapon/W, mob/user)
+	if(W.is_screwdriver(user))
+		to_chat(user, "<span class = 'warning'>The timer seems locked to three seconds.</span>")
+		return
+	..()
+
+/obj/item/weapon/grenade/holy/prime()
```

Comment:
Does prime() get called after the detonation time is up?

Reply:
yes